### PR TITLE
:bug: clean up metric resource when disable metrics

### PR DESCRIPTION
**Pull request by ldpliu**
## Related issue(s)

Fixes #

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/manifests/grafana/acm-global-postgres-exporter.yaml [hunk 1]
```diff
@@ -3280,4 +3280,6 @@ kind: ConfigMap
 metadata:
   name: grafana-dashboard-acm-global-postgres-exporter
   namespace: {{ .Namespace }}
+  labels:
+    dependon: enableMetrics
```
Comment (yanmxa) [inline operator/pkg/controllers/hubofhubs/manifests/grafana/acm-global-postgres-exporter.yaml:3284]: Can we align the owner label with other components key: `global-hub.open-cluster-management.io/managed-by`. And then just define the value like `operand-metrics`, `global-hub-metrics`?

Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/manifests/grafana/acm-global-postgres-exporter.yaml:3284]: > Can we align the owner label with other components key: `global-hub.open-cluster-management.io/managed-by`. And then just define the value like `operand-metrics`, `global-hub-metrics`?

As they already have the `global-hub.open-cluster-management.io/managed-by:global-hub-operator` label and it's meaningful. So I think use a new label to identity the metrics  is better. What do you think?

Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/manifests/grafana/acm-global-postgres-exporter.yaml:3284]: I mean the key `global-hub.open-cluster-management.io/managed-by` is used for owner relationships. so we can just use the key and give another value name for these metrics resources?

File: operator/pkg/controllers/hubofhubs/globalhub_controller.go [hunk 2]
```diff
@@ -216,6 +216,13 @@ func (r *MulticlusterGlobalHubReconciler) Reconcile(ctx context.Context, req ctr
 		klog.Info("Kafka and Postgres are provided by customer, disable metrics")
 	}
 
+	if !mgh.Spec.EnableMetrics {
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/globalhub_controller.go:219]: maybe we define a new method `reconcileMetrics` to include all related with enableMetrics.

File: pkg/constants/constants.go [hunk 3]
```diff
@@ -79,7 +79,9 @@ const (
 	// Deprecated identify the resource is a local-resource
 	// GlobalHubLocalResource = "global-hub.open-cluster-management.io/local-resource"
 	// if the resource with this label, it will be synced to database and then propagated to managed hub
-	GlobalHubGlobalResourceLabel = "global-hub.open-cluster-management.io/global-resource"
+	GlobalHubGlobalResourceLabel   = "global-hub.open-cluster-management.io/global-resource"
+	GlobalhubDependOnLabelKey      = "dependon"
```
Comment (clyang82) [inline pkg/constants/constants.go:83]: the label name should have this prefix `global-hub.open-cluster-management.io/`. so you can define the label name as `global-hub.open-cluster-management.io/metrics-resource`

BTW, you need to update this doc to reflect your changes - https://github.com/stolostron/multicluster-global-hub/tree/main/doc/reserved_names

Comment (ldpliu) [reply pkg/constants/constants.go]: Done

File: operator/pkg/controllers/hubofhubs/globalhub_controller.go (hunk 2 above)
Comment (ldpliu) [reply operator/pkg/controllers/hubofhubs/globalhub_controller.go]: Done

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**